### PR TITLE
Updates for Kubernetes 1.8

### LINKED ISSUE #63: Kube 1.8 busted -- wrong column getting "clusterip" for gluster storage class
https://github.com/redhat-nfvpe/kube-centos-ansible/blob/2fce185a5a88d90787df1ebfe8dada7112b571ee/roles/glusterfs-kube-config/tasks/main.yml#L89

---

```
[centos@kube-master ~]$ kubectl get svc --namespace=gluster | grep 8080 | awk '{print $2}'
ClusterIP
[centos@kube-master ~]$ kubectl get svc --namespace=gluster | grep 8080 | awk '{print $3}'
10.97.143.109

```

### PR DESCRIPTION
**Pull request by dougbtv**
Short list is...

* Update `kube_version` variable to accept `latest` as a value, default to that.
* Runs `kubectl proxy` to allow API access
* Query API to figure out Heketi IP address
* Properly set Heketi IP address given that.

Tested it pretty heavily. Both with latest and backwards with 1.7.5.

Fixes #63

### CODE REVIEW

File: roles/glusterfs-kube-config/tasks/main.yml [hunk 1]
```diff
@@ -91,11 +92,19 @@
...
     state: directory
 
-- name: Get the heketi IP address
+- name: Query the API for the heketi IP address
   shell: >
-    kubectl get svc --namespace=gluster | grep 8080 | awk '{print $2}'
+    curl -s localhost:{{ kubectl_proxy_port }}/api/v1/namespaces/gluster/services/heketi | jq -r .spec.clusterIP
   register: heketi_ipaddress_raw
 
+- name: Fail when there's no IP address (this could use grooming)
```
Comment (leifmadsen) [inline roles/glusterfs-kube-config/tasks/main.yml:100]: Might want to open a low priority issue for this then

File: roles/kubectl-proxy-systemd/default/main.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+---
```
Comment (leifmadsen) [inline roles/kubectl-proxy-systemd/default/main.yml]: do you really want a blank file here?

File: roles/kubectl-proxy-systemd/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    mode: 0755
+
+- name: Template kubectl proxy systemd unit
+  template:
+    src: kubectl-proxy.service.j2
+    dest: /etc/systemd/system/kubectl-proxy.service
+  register: template_kubectlproxy_unit
+
+- name: Systemctl daemon-reload when appropriate
+  shell: >
```
Comment (leifmadsen) [inline roles/kubectl-proxy-systemd/tasks/main.yml]: does the systemd module do this function now?

**Review by leifmadsen — COMMENTED** [review]
Mostly approved. Not sure about that blank file though :)

File: roles/kubectl-proxy-systemd/default/main.yml (hunk 2 above)
Comment (dougbtv) [reply roles/kubectl-proxy-systemd/default/main.yml]: Nope! good catch.

File: roles/kubectl-proxy-systemd/tasks/main.yml (hunk 3 above)
Comment (dougbtv) [reply roles/kubectl-proxy-systemd/tasks/main.yml]: indeed it does, thanks.

**Review by leifmadsen — APPROVED** [review]
:shipit:

**Pull request merged**